### DISCUSSION TITLE
Implement Lods(Level of Details) for filament

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4413] [Q&A]
I need implement lod for my project recently and can't wait.So I need implement LODs by myself.
Can some one give me some suggestion for Lods, or some code  snippets for implement LODs for filament.

Thanks for any help.

**Comment by jeanlemotan** [comment]
+1 for LODs
For this it would be nice to use the ratio of the projected bounding sphere and viewport diagonal instead of the distance, like this it requires less tuning based on the object world size. It's also cheap - just a dot product and a division with a loop constant (the diagonal).
The thresholds then are specified in percentage (10% of the screen size -> LOD2, 5% -> LOD3 etc)

**Reply by romainguy** [reply]
Our thinking it that we would use the projected area in pixels or screen ratio instead of the distance indeed. We decided long ago that distance is a not a good metric as it depends too much on the resolution, the size of the object, etc.
